src/system/tests/TestSystemScheduleLambda.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <system/SystemConfig.h>
+
+#include <lib/support/CodeUtils.h>
+#include <lib/support/ErrorStr.h>
+#include <lib/support/UnitTestRegistration.h>
+#include <nlunit-test.h>
+#include <platform/CHIPDeviceLayer.h>
+
+// Test input data.
+
+static void CheckScheduleLambda(nlTestSuite * inSuite, void * aContext)
+{
+    bool * called = new bool(false);
+#if CHIP_SYSTEM_CONFIG_USE_DISPATCH
+    chip::DeviceLayer::PlatformMgr().RunEventLoop();
+    chip::DeviceLayer::SystemLayer().ScheduleLambda([called] {
```

Comment:
This is obviously not going to work (and doesn't), since RunEventLoop is a blocking call that does not return until StopEventLoopTask is called.

If you fix this code to be right, I would expect you don't need to do this branching on USE_DISPATCH or this public ProcessDeviceEvents thing.  This should work across all our platforms:
```
    chip::DeviceLayer::SystemLayer().ScheduleLambda([called] {
        *called = true;
        chip::DeviceLayer::PlatformMgr().StopEventLoopTask();
    });
    chip::DeviceLayer::PlatformMgr().RunEventLoop();
```

Reply:
I tried that, `RunEventLoop` on FreeRTOS requires `StartEventLoopTask` to work, and lots of extra initialize codes, I'm unable make it work for esp32.